lib/compat/wordpress-6.3/behaviors.php

```diff
@@ -0,0 +1,19 @@
+<?php
+/**
+ * Behaviors.
+ *
+ * @package gutenberg
+ */
+
+add_filter(
+	'block_editor_settings_all',
+	function( $settings ) {
+		$theme_data = WP_Theme_JSON_Resolver_Gutenberg::get_theme_data()->get_data();
+		if ( array_key_exists( 'behaviors', $theme_data ) ) {
+			$settings['behaviors'] = $theme_data['behaviors'];
+		}
+		// TODO: Make sure to also get the value from the core theme.json file.
+		return $settings;
+	},
+	PHP_INT_MAX
```

Comment:
is this for development only? if so maybe let's add a `@TODO` to remove this before merge.

Reply:
yes, this is still a TODO to be done before merging 👍

Reply:
FYI, I've now moved the PHP code specific to the behaviors to [`block-editor-settings.php`](https://github.com/WordPress/gutenberg/blob/fc812bc6c04e8b3c5c7fe3450c1e39d1dd6f6f3b/lib/block-editor-settings.php)

Reply:
FYI 2: I've reverted the change from above because I've realized that `block-editor-settings.php` is for GB-plugin-specific stuff only. Sorry for the complication 🙂

Reply:
The current priority prevents other plugins from updating the `behaviors` fields in the settings passed to the client.

Reply:
I see, thanks for pointing this out Greg. Will fix it!

Reply:
Just to clarify, will fix it in a follow-up PR most probably 👍